### ISSUE TITLE
Cannot tell what element owns an attribute

### ISSUE THREAD

**Issue by mjgp2** [w3c/dom#6]
There is no way to link an attribute (`Attr`[1]) back to the element it is owned by - this seems a gap in the spec, as, for example, if you return `Attr` objects from an xpath query they do not contain any context to their location in the DOM.

Indeed, the spec itself says:

> Attributes have a namespace (null or a non-empty string), namespace prefix (null or a non-empty string), local name (a non-empty string), name (a non-empty string), value (a string), and element (null or an element).

However the IDL does not contain the "element" referenced in the description:

```
[Exposed=Window]
interface Attr {
  readonly attribute DOMString? namespaceURI;
  readonly attribute DOMString? prefix;
  readonly attribute DOMString localName;
  readonly attribute DOMString name;
           attribute DOMString value;
  readonly attribute boolean specified; // useless; always returns true
};
```

There seems to have been some thought about this in the past; for the MDN page[2]:

```
ownerElement Deprecated since Gecko 7.0 Obsolete since Gecko 29.0
This property has been removed from Firefox 29. Since you can only get Attr objects from elements, you should already know the owner.
Contrary to above claim, Document.evaluate can return Attr objects from an XPath, in which case you would not easily know the owner.
```

Firefox and Chrome still have `ownerElement` as a property of `Attr`. Can I suggest that it is added (back?) into the spec?

This is a real issue that must be addressed before the spec is finalised.

Many thanks

[1] https://www.w3.org/TR/dom/#interface-attr
[2] https://developer.mozilla.org/en/docs/Web/API/Attr

**Comment by ArkadiuszMichalski** [comment]
@mjgp2 You should not read this spec because it's old snapshot of https://dom.spec.whatwg.org/ and and contains a mass of gaps.

**Comment by yongsheng** [comment]
since firefox and chrome both have this property, I'll bring it back in 4.1.

**Comment by bedney** [comment]
This is critical to the way some of our functionality works. Whether or not the Attr API got completely redone, and is no longer a Node or whatever, the fact that I can have a handle to an Attr 'object' and ask it 'what Element owns you?' is the salient part. Please restore this.

**Comment by yongsheng** [comment]
There will be 7 milestones , see below. Actually the first working draft doesn't have so many changes except bs format and one typo fix. will fix all problems in the following milestones.
1.15 C 1st working draft
2.26 C 2nd round
4.9 C 3rd round
5.21 C 4th round
6.31 C 5th round 
7.30 - DOM 4.1 working draft, finalization

**Comment by bedney** [comment]
Ah, ok - I was unaware of this schedule. Thank you very much!

**Comment by yongsheng** [comment]
This fix will be included in the 3rd round version.

**Comment by yongsheng** [comment]
It'll be fixed together with #77
